File: rpnumerics/physics/FluxFunction.java
/*
 * Instituto de Matematica Pura e Aplicada - IMPA
 * Departamento de Dinamica dos Fluidos
 *
 */

package rpnumerics.physics;

import wave.util.RealVector;
import wave.util.RealMatrix2;
import wave.util.HessianMatrix;

public  interface FluxFunction
{
	HessianMatrix D2F( RealVector U );
	
	RealMatrix2 DF( RealVector U );
	
	RealVector F( RealVector U );
	
	FluxParams fluxParams( );
	
	
}

File: src/rpnumerics/FluxFunction.cc
#include "FluxFunction.h"

FluxFunction::FluxFunction(const FluxParams & params):params_(new FluxParams(params)){}

FluxFunction::~FluxFunction(void) {
    delete params_;
}


const FluxParams & FluxFunction::fluxParams(void) const {
    return *params_;
}

void FluxFunction::fluxParams(const FluxParams & params) {
    delete params_;
    params_=new FluxParams(params.params());
    
}
